tests/src/main/java/net/neoforged/neoforge/debug/entity/EntityEventTests.java

```diff
@@ -69,4 +71,19 @@ static void entityAttributeModificationEvent(final DynamicTest test, final Regis
                         donkey, d -> d.getAttribute(testAttr.get()).getValue(), "test attribute", 1.5D))
                 .thenSucceed());
     }
+
+    @GameTest
+    @EmptyTemplate(floor = true)
+    @TestHolder(description = "testing the event by converting a nitwit into a butcher", enabledByDefault = true)
+    static void villagerChangeProfessionEvent(final DynamicTest test) {
+        test.eventListeners().forge().addListener((final VillagerChangeProfessionEvent event) -> {
+            if (event.getOldProfession() == event.getNewProfession())
+                return;
+
+            if (event.getOldProfession() == VillagerProfession.FLETCHER) {
+                event.setNewProfession(VillagerProfession.BUTCHER);
+            }
+            test.pass();
+        });
+    }
```

Comment:
I think this test needs to be improved. The test should involve summoning a villager, changing its profession, and seeing if the event listener is trigger and if the profession is changed as expected. Maybe also a second test for cancelling the profession change should be made. @Matyrobbrt, any suggestions/recommendations on this front?

Reply:
I'm a noob when it comes to the testframework, so it's been a pain in the arse.
Yes, if @Matyrobbrt could suggest anything, please do.